src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -271,7 +272,7 @@ class MarathonSchedulerActor(
     withLockFor(Set(appId))(f)
 
   // there has to be a better way...
-  def driver: SchedulerDriver = MarathonSchedulerDriver.driver.get
+  def driver: SchedulerDriver = marathonSchedulerDriverHolder.driver.get
```

Comment:
Ah got it. Not obvious looking at the code. Perhaps make a small note on the holder class?